core/server/common/src/main/java/alluxio/master/metastore/BlockStore.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.metastore;
+
+import alluxio.master.metastore.BlockStore.Block;
+import alluxio.proto.meta.Block.BlockLocation;
+import alluxio.proto.meta.Block.BlockMeta;
+
+import java.util.List;
+import java.util.Optional;
+
+import javax.annotation.concurrent.NotThreadSafe;
+
+/**
+ * The block store keeps track of block sizes and block locations.
+ *
+ * BlockStores require external synchronization for operations on the same block id. Operations on
+ * different block ids can be performed concurrently.
+ */
+@NotThreadSafe
+public interface BlockStore extends Iterable<Block> {
+  /**
+   * @param id a block id
+   * @return the block's metadata, or empty if the block does not exist
+   */
+  Optional<BlockMeta> getBlock(long id);
```

Comment:
We previously experimented with optional when building the worker metastore and found it was too verbose. Do we expect to be calling `getBlock` on blocks which do not exist?

Reply:
To check whether a block exists, we use 
```java
blockStore.getBlock(blockId).isPresent
```
To check whether a block exists, and then use its metadata if it does, we use
```java
Optional<BlockMeta> block = blockStore.getBlock(blockId);
if (!block.isPresent()) {
  // handling
}
// do things with block.get(), or save it to a local variable to use it many times.
```